### PR TITLE
Create argocd pull model installation manifest

### PR DESCRIPTION
**Pull request by lennysgarage**
This PR is for creating the AppLC argocd pull model installation manifests. Adding three new controllers,  `argocd-pull-integration-controller-manager`, `multicluster-integrations-syncresource`, and `multicluster-integrations-aggregation`. Also a new `multiclusterapplicationsetreport` crd.

https://issues.redhat.com/browse/ACM-3404

Signed-off-by: Jonathan Marcantonio <jmarcant@redhat.com>

### CODE REVIEW

File: pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                operator: In
+                values:
+                - linux
+      containers:
+      - args:
+        - --secure-listen-address=0.0.0.0:8443
+        - --upstream=http://127.0.0.1:8080/
+        - --logtostderr=true
+        - --v=0
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.13.0
```
Comment (joeg-pro) [inline pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml:48]: I think there will be some broader discussion about this PR, but I was just glancing at this and see this hardcoded image ref to something coming out of Google container registry.  This will not be acceptable for the productized ACM.  All images must be productized ones served out of registry.redhat.io.  Besides that, the image will have to appear in the image manifest, be referenced by an image override, etc.

File: pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      creationTimestamp: null
+      labels:
+        control-plane: controller-manager
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
```
Comment (joeg-pro) [inline pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml:28]: Why is this required-during-scheduling node affinity spec necessary here?  I have start to see something similar in some other non-ACM things, but we do not customarily include such a thing in ACM or MCE deployments....wondering where it comes from, what the use case is that requires it, and if its somethign we should be adding across the board for ACM and MCE if the use case is relevant to us.

Comment (mikeshng) [reply pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml:28]: Thanks for the review Joe. We will remove this additional deployment.
This came from the kubebuilder project bootstrap.

File: pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml (hunk 1 above)
Comment (mikeshng) [reply pkg/templates/charts/toggle/multicloud-operators-subscription/templates/argocd-pull-integration-controller-manager.yaml:48]: Thanks for the review Joe. We will remove this additional deployment.
This came from the kubebuilder project bootstrap.

**Comment by xiangjingli** [comment]
@joeg-pro @JakobGray The PR has been modified based on Joe's comments. Could you review it again when you have a cycle. 

As discussed in the Sos, We will create a user story for involving install squad as early as possible next time when we start a new project

**Comment by xiangjingli** [comment]
The operator bundle has been updated by another PR created by automation, closing ...
https://github.com/stolostron/multiclusterhub-operator/pull/981
